src/gui/processing/qgsprocessingwidgetwrapperimpl.cpp

```diff
@@ -4083,28 +4094,42 @@ void QgsProcessingFieldWidgetWrapper::setParentLayerWrapperValue( const QgsAbstr
     QgsProcessingFeatureSourceDefinition fromVar = qvariant_cast<QgsProcessingFeatureSourceDefinition>( value );
     value = fromVar.source;
   }
-  QgsVectorLayer *layer = QgsProcessingParameters::parameterAsVectorLayer( parentWrapper->parameterDefinition(), value, *context );
-  if ( layer && layer->isValid() )
+
+  bool valueSet = false;
+  QList< QgsMapLayer * > layers = QgsProcessingParameters::parameterAsLayerList( parentWrapper->parameterDefinition(), value, *context );
+  if ( !layers.isEmpty() )
```

Comment:
I'd prefer if the original code was left whenever layers contains only a single layer -- the original code has the benefit that it sets the layer for the field combo box, which means that the field list dynamically responds to changes in the layer structure (fields added, removed, renamed...)

Reply:
Sure, it's done